pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -246,15 +345,24 @@ func (c *BootstrapTeardownController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
-// this function checks if etcd has scaled the initial bootstrap cluster
-// with all 3 masters
-func (c *BootstrapTeardownController) isEtcdAvailable(currentEtcdOperator *operatorv1.Etcd) bool {
-	if operatorv1helpers.IsOperatorConditionTrue(currentEtcdOperator.Status.Conditions, clustermembercontroller.ConditionBootstrapSafeToRemove) {
-		klog.V(2).Info("etcd is available, bootstrap complete")
-		return true
+func (c *BootstrapTeardownController) hasMoreThanTwoEtcdMembers() (bool, error) {
+	cli, err := c.getEtcdClient()
+	if err != nil {
+		return false, err
+	}
+	defer cli.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	l, err := cli.MemberList(ctx)
+	cancel()
+	if err != nil {
+		return false, err
 	}
-	c.eventRecorder.Eventf("WaitingForEtcdToBeAvailable", "waiting for %s to be true", clustermembercontroller.ConditionBootstrapSafeToRemove)
-	return false
+
+	if len(l.Members) > 2 {
```

Comment:
This condition is not enough, we not only need to check for 2 members but also need to check if both of them are healthy.

Reply:
> This condition is not enough, we not only need to check for 2 members but also need to check if both of them are healthy.
> /cc @hexfusion

if we add one at a time, this is better than where we are.

Where can I see health?

Reply:
```
	for _, m := range l.Members {
		statusResp, err := cli.Status(context.Background(), m.ClientURLs[0])
		if err != nil {
			klog.Warningf("member %s with etcdName %s is not healthy", m.Name,
				getEtcdName(m.PeerURLs))
			klog.Errorf("error reading %s health: %#v", m.Name, err)
			continue
		}
		klog.Infof("member %s is healtly with %d index", m.Name, statusResp.RaftIndex)
        }
```

whithout this the member remove calls will fail, but I can see how temporary we will go to degraded and as soon as 2nd member is healthy the remove calls will succeed and well not be degraded anymore!